lib/model.js

```diff
@@ -233,50 +281,13 @@ const _create = (data, definition) => {
   };
 
   definition.methods.save = async function(options) {
-    if (!this.isNew() && !this.isModified()) {
-      return this;
-    }
-    let obj = this.toObject();
-    if ('function' === typeof this.beforeSave) {
-      await this.beforeSave.apply(obj);
-    }
-    const value = await this._validate(obj, options);
-    obj = EJSON.serialize(value);
-    const original = obj.__original;
-    delete obj.__modified;
-    delete obj.__original;
-    const newEtag = etag(JSON.stringify(obj));
-    const collection = definition.collection;
-    let newOrUpdatedModel;
-    if (!this.isNew()) {
-      const query = { _id: obj._id, _etag: obj._etag };
-      obj._etag = newEtag;
-      debug(`updating ${collection}: ${JSON.stringify(obj)}`);
-      const result = await db.replace(collection, query, obj);
-      if (result.matchedCount === 0) {
-        throw errors.Conflict({ message: 'Document update conflict' });
-      }
-      newOrUpdatedModel = this.set({
-        _etag: newEtag,
-        __modified: [],
-        __original: null
-      }, { silent: true });
-    } else {
-      obj._etag = newEtag;
-      debug(`inserting into ${collection}: ${JSON.stringify(obj)}`);
-      const result = await db.insert(collection, obj);
-      debug(`insert into ${collection} successful`, result);
-      obj = Array.isArray(result.ops) ? result.ops[0] : result.ops;
-      newOrUpdatedModel = _create(obj, definition);
-    }
-    if ('function' === typeof newOrUpdatedModel.afterSave) {
-      // should not be awaited
-      const afterSave = newOrUpdatedModel.afterSave(original || null);
-      if (pathOr(false, ['waitAfterSave'], options)) {
-        await afterSave;
-      }
-    }
-    return newOrUpdatedModel;
+    options = Object.assign(options || {}, { canPassID: false });
+    return _saveHelper(this, options);
+  };
+
+  definition.methods.saveWithId = async function(id, options) {
+    options = Object.assign(options || {}, { canPassID: true, _id: id });
```

Comment:
I think we want to throw an error with this function if the object is not new? right now, the id will just be ignored